server.js

```diff
@@ -0,0 +1,50 @@
+'use strict';
+const net = require('net');
+const server = net.createServer();
+const parse = require('./lib/parse-message.js');
+
+let sockets = [];
+
+server.on('connection', function(socket){
+  console.log('a client connected');
+  socket.write('welcome to wack chat');
+  socket.username = `user_${Math.floor(Math.random()*100)}`;
+  sockets.push(socket);
+
+  socket.on('data', function(buffer){
+    console.log(buffer.toString());
+    let message = buffer.toString();
+    if(message.startsWith('/nick'))
+      return parse.nickCommand(message, socket);
+
+    if(message.startsWith('/dm'))
+      return parse.dmCommand(message, sockets);
+
+    if(message.startsWith('/user'))
+      return parse.userCommand(sockets, socket);
+
+    if(message.startsWith('/troll'))
+      return parse.trollCommand(message, sockets);
+
+    if(message.startsWith('/ban'))
+      return parse.banCommand(message, sockets);
+
+    sockets.forEach(s => {
```

Comment:
Not a point-loser, but this ``forEach`` could be taken out since it's not a great idea to broadcast things to everyone by default from a UI experience.